datasets/flwr_datasets/partitioner/inner_dirichlet_partitioner.py

```diff
@@ -129,6 +128,17 @@ def num_partitions(self) -> int:
         self._determine_partition_id_to_indices_if_needed()
         return self._num_partitions
 
+    @property
+    def partition_id_to_indices(self) -> Dict[int, List[int]]:
```

Comment:
I wonder if it's worth it moving most of this logic to a separate aux method that's also called by `num_partitions()`, `load_partition()`. The code is repeated (except for the `return` statement)

Reply:
This `3x` repetition happens also in `shard_`, `inner_dirichtlet_`, `dirichlet_`, and `natural_id` partitioners.